phpBB/includes/functions_admin.php

```diff
@@ -2853,35 +2851,25 @@ function get_database_size()
 			{
 				$version = $row['mysql_version'];
 
-				if (preg_match('#(3\.23|[45]\.|10\.[0-9]\.[0-9]{1,2}-+Maria)#', $version))
-				{
-					$db_name = (preg_match('#^(?:3\.23\.(?:[6-9]|[1-9]{2}))|[45]\.|10\.[0-9]\.[0-9]{1,2}-+Maria#', $version)) ? "`{$db->get_db_name()}`" : $db->get_db_name();
+				$db_name = $db->get_db_name();
 
-					$sql = 'SHOW TABLE STATUS
-						FROM ' . $db_name;
-					$result = $db->sql_query($sql, 7200);
+				$sql = 'SHOW TABLE STATUS
+					FROM ' . $db_name;
+				$result = $db->sql_query($sql, 7200);
 
-					$database_size = 0;
-					while ($row = $db->sql_fetchrow($result))
+				while ($row = $db->sql_fetchrow($result))
+				{
+					if (isset($row['Engine'])
+						&& ($row['Engine'] == ('MyISAM' || 'InnoDB' || 'Aria'))
```

Comment:
Very interesting comparison for php

Reply:
Got confused by `arbitrary expressions in empty`, my bad.